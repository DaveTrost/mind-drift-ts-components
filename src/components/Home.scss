.Home {

  .homeContainer {
    text-align: center;
    margin: 0;
    margin-top: 15%;
  }

  .logo {
    height: 25vh;
    width: 60%;
    max-width: 420px;
    margin: auto;
    margin-bottom: 10%;
    img {
      margin: 0;
      width: 50%;
      height: auto;
    }
    @media screen and (min-width: 425px) {
      img {
        width: 35%;
      }
    }
    @media screen and (min-width: 1440) {
      img {
        width: 30%;
      }
    }
  }

  .headerText {
    display: flex;
    flex-direction: row;
    justify-content: center;
    h2 {
      font-family: Montserrat;
      font-weight: 200;
      font-size: 1.5em;
      line-height: 1.6em;
      color: var(--dark-blue);
    }
    h3 {
      font-family: Damion;
      font-weight: 700;
      font-size: 1.9em;
      color: var(--peach);
    }
  }
  
  .linksContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-decoration: none;
    a {
      text-decoration: none;
      color: var(--dark-blue);
    }
  }

  .breatheButton, .setupButton {
    display: block;
    border: solid var(--light-blue) 4px;
    border-radius: 15px;
    background-color: white;
    width: 15em;
    height: 3em;
    margin: auto;
    margin-top: 1em;
    color: var(--dark-blue);
    font-family: Montserrat;
    font-weight: 400;
    font-size: 14px;
  
    &:hover {
      border: none;
      background-color: var(--light-blue);
      color: white;
      cursor: pointer;
      text-decoration: none;
    }
  }
}
