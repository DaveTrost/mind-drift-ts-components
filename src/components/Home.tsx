import React from 'react';
import { Link } from 'react-router-dom';
import NavBar from './NavBar';
// import AchievementToast from './AchievementToast';
import mindDriftIcon from '../assets/mindDriftIcon.png';
import './Home.scss';

const Home = () => (
  <section className="Home">
    <NavBar />
    {/* <AchievementToast /> */}
    <div className="homeContainer">
      <div className="logo">
        <img src={mindDriftIcon} alt="mindDrift logo" />
        <section className="headerText">
          <h2>mind</h2>
          <h3>Drift</h3>
        </section>
      </div>
      <div className="linksContainer">
        <Link to="/breathe">
          <button className="breatheButton">Breathe</button>
        </Link>
        <Link to="/settings">
          <button className="setupButton">Setup</button>
        </Link>
      </div>
    </div>
  </section>
);

export default Home;
