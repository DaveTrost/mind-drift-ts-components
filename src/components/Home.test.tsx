import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('Home component', () => {
  it('renders the Home component', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
