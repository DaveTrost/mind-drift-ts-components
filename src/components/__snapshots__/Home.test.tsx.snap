// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders the Home component 1`] = `
<section
  className="Home"
>
  <NavBar />
  <div
    className="homeContainer"
  >
    <div
      className="logo"
    >
      <img
        alt="mindDrift logo"
        src="test-file-stub"
      />
      <section
        className="headerText"
      >
        <h2>
          mind
        </h2>
        <h3>
          Drift
        </h3>
      </section>
    </div>
    <div
      className="linksContainer"
    >
      <Link
        to="/breathe"
      >
        <button
          className="breatheButton"
        >
          Breathe
        </button>
      </Link>
      <Link
        to="/settings"
      >
        <button
          className="setupButton"
        >
          Setup
        </button>
      </Link>
    </div>
  </div>
</section>
`;
